lms/envs/common.py

```diff
@@ -1041,15 +1041,10 @@
 )
 
 # Include JS files needed for courseware search
-courseware_js += (
-    'js/search/models/result.js',
-    'js/search/collections/collection.js',
-    'js/search/views/form.js',
-    'js/search/views/list.js',
-    'js/search/views/item.js',
-    'js/search/router.js',
-    'js/search/app.js',
-)
+if FEATURES.get('ENABLE_COURSEWARE_SEARCH'):
```

Comment:
Can't make this conditional here, because ENABLE_COURSEWARE_SEARCH is defined as False within this file. So, the processing of the settings overrides from json or other location have not yet been processed yet.